### DISCUSSION TITLE
How to raise Error on LiteralString

### DISCUSSION THREAD

**Discussion by randolf-scholz** [python/typing#1434] [Q&A] [answered]
This issue come from [pandas-stubs](https://github.com/pandas-dev/pandas-stubs/pull/756).

Is it possible, using `LiteralString` [PEP675](https://peps.python.org/pep-0675/), to write an overloaded function `foo(x)` such that

- `foo(x: Literal["a", "b", "c"])` passes
- `foo(x: LiteralString)` raises a typing error
- `foo(x: str)` passes

i.e. any non-literal string is ok, but for literal string the type checking acts as a defense against typos.

**Comment by erictraut** [comment, marked as answer]
No, that's not possible (nor would it make sense given the way that subtyping and overloads work in Python). `LiteralString` is a subtype of `str`, so if you provide an overload that accepts a `str`, it will (by definition) accept a `LiteralString`. A type checker will emit an error only if the supplied arguments fail to match any of the overloaded signatures.

**Reply by randolf-scholz** [reply]
So I guess this is another case where one would love to have `Intersection` and `Not`, since then `Intersection[str, Not[LiteralString]]` would be the correct overload.

**Reply by Akuli** [reply]
A `@typing_error` decorator would help too: https://github.com/python/typing/issues/1043
